engine/src/conversion/analysis/fun/mod.rs

```diff
@@ -192,25 +191,40 @@ impl Default for ReturnTypeAnalysis {
     }
 }
 
+pub(crate) struct PodAndConstructorAnalysis {
+    pub(crate) pod: PodAnalysis,
+    pub(crate) constructors: PublicConstructors,
+}
+
 /// An analysis phase where we've analyzed each function, but
 /// haven't yet determined which constructors/etc. belong to each type.
 pub(crate) struct FnPrePhase;
 
 impl AnalysisPhase for FnPrePhase {
     type TypedefAnalysis = TypedefAnalysis;
-    type StructAnalysis = PodAnalysis;
+    type StructAnalysis = PodAndConstructorAnalysis;
     type FunAnalysis = FnAnalysis;
 }
 
 pub(crate) struct PodAndDepAnalysis {
     pub(crate) pod: PodAnalysis,
     pub(crate) constructor_and_allocator_deps: Vec<QualifiedName>,
+    pub(crate) constructors: PublicConstructors,
 }
 
 /// Analysis phase after we've finished analyzing functions and determined
 /// which constructors etc. belong to them.
 pub(crate) struct FnPhase;
 
+/// Indicates which kinds of public constructors are known to exist for a type.
+#[derive(Debug, Default, Copy, Clone)]
+pub(crate) struct PublicConstructors {
+    pub(crate) default_constructor: bool,
+    pub(crate) copy_constructor: bool,
```

Comment:
As far as I can see, a couple of these fields are never read.

When I merge this with #879 this gives me compile-time warnings. I'm not sure why it doesn't here.

Reply:
I think it's just this one, which ended up not being referenced by other stages. Done